rpc/codec/conn.go

```diff
@@ -9,94 +9,50 @@ import (
 	"encoding/binary"
 	"io"
 	"net"
-	"sync"
-
-	"code.google.com/p/snappy-go/snappy"
 
+	wire "github.com/cockroachdb/cockroach/rpc/codec/wire.pb"
 	"github.com/gogo/protobuf/proto"
 )
 
-type frameBuf struct {
-	size [binary.MaxVarintLen64]byte
+const enableSnappy = true
+
+type decompressFunc func(src []byte, m proto.Message) error
+
+var decompressors = [...]decompressFunc{
```

Comment:
Hmm, I saw this done somewhere else. Seems like the "keys" are indexes in the array and do have an effect. See https://play.golang.org/p/6YTkeXjVAc.

It is true that the current usage of `decompressors` will not gracefully handle the reception of an unknown enum value, but that can be fixed.